### PR TITLE
Introduce library

### PR DESCRIPTION
**Pull request by makimenko**
- @ComFreek what you do you think about splitting the code into into library and application? Just thinking if that's make sense.
- "atft" stands for Angular Template for Threejs. Any other name is welcome.
- NOTE: Initially I was not able to watch the library changes during ng serve. But found the workaround to change in app.module.ts import to:
  ```
  import { AtftModule } from 'projects/atft/src/lib/atft.module';
  ```

### CODE REVIEW

**Comment by ComFreek** [comment]
> what you do you think about splitting the code into into library and application?

Didn't we have that before and reverted it? I vaguely remember there to be some small annoyance from Angular's or Three.js's side. In any case, I welcome the separation -- if it's implementable.

> "atft" stands for Angular Template for Threejs. Any other name is welcome.

`atft` is short & nifty -- but also not self-explanatory. What about `angular-three`? I am not sure if names for 3rd-party libraries including `angular` and `three` are encouraged. E.g. PHP discourages that. However, I've seen a lot of libraries with `angular-` prefix.

I'll have a look at the code.

File: projects/atft/karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Karma configuration file, see link for more information
```
Comment (ComFreek) [inline projects/atft/karma.conf.js]: Let's see if that works with the e2e tests. I had to adjust the previous `karma.conf.js` to be able to run the e2e tests, also on Travis.

**Comment by ComFreek** [comment]
Oh, now I see why a short name is preferable. Indeed, if we use it for custom DOM elements as a prefix, we should keep it short. [Angular Google Maps](https://angular-maps.com/) also a short name: "agm".

The (non-autogenerated :smile:) code looks good!

How about publishing `atft` on NPM as a stand-alone library without examples? Then the workflow for users would be: i) npm install atft, ii) change app.module.ts iii) ready to go!

[complete-beginner-guide-to-publish-an-angular-library-to-npm](https://medium.com/@esanjiv/complete-beginner-guide-to-publish-an-angular-library-to-npm-d42343801660) might be relevant.

**Comment by ComFreek** [comment]
I am working on a package.json, hang on :smile:

File: projects/atft/package.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{  
+  "name": "atft",
+  "version": "0.0.1",
+  "description": "Ready-to-use Angular components for Three.js, see https: //makimenko.github.io/angular-template-for-threejs/.",
```
Comment (ComFreek) [inline projects/atft/package.json]: @makimenko You might want to tweak the description a bit -- if you want.

File: projects/atft/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    {
+      "name": "ComFreek",
+      "url": "https: //github.com/ComFreek"
+    }
+  ],
+  "peerDependencies": {
+    "@angular/common": "^7.2.0",
+    "@angular/core": "^7.2.0",
+    "three": "^0.103.0"
```
Comment (ComFreek) [inline projects/atft/package.json]: Apparently, one should duplicate dependencies from the outer `package.json` in this `package.json`: https://stackoverflow.com/questions/50189589/angular-cli-6-where-to-put-library-dependencies. Hence, I put `three` in there.

**Comment by makimenko** [comment]
Thanks for help!

> Didn't we have that before and reverted it? 

Yep, in "three" <=0.99 there were no examples as modules and the only way how to enable examples/js was to import Java Script file (import ../js/EnableThreeExamples.js). Nevertheless, with such approach it was not possible to create separate library - it was failing during compilation.
In recent version I found that team implemented examples in separate "jsm" folder (as Module). Looks great and seems working fine and much cleaner.

> Or was is due to the "watch" issue you now have resolved?

Indeed. That was missing previously as well. 
Yesterday, found a workaround on internet (import not a library, but module via patch). During development it is nice to use auto-watch feature (when needed).

> Let's see if that works with the e2e tests. I had to adjust the previous karma.conf.js to be able to run the e2e tests, also on Travis.

Thanks.

> Oh, now I see why a short name is preferable. Indeed, if we use it for custom DOM elements as a prefix, we should keep it short.

Ok

> How about publishing atft on NPM as a stand-alone library without examples? 
> complete-beginner-guide-to-publish-an-angular-library-to-npm might be relevant.

Yep, that's a good idea: publish library to npm.
Let's do in a separate Issue/PR.

**Comment by makimenko** [comment]
> Codacy/PR Quality Review

Seems this is not suits for this project. Will switch to more appropriate app later.

**Comment by ComFreek** [comment]
> Codacy/PR Quality Review — Not up to standards. This pull request quality could be better. 

I don't know why [it complains about not using double quotes](https://app.codacy.com/app/mihails.akimenko/angular-template-for-threejs/pullRequest?prid=3408731). The tslint.json in project/atft in fact includes the root ./tslint.json one. And the latter specifies single quotes as the standard. I guess this is some kind of bug with Codacy. Also, running `tslint` manually doesn't show me any errors.

> codecov/project — 60.26% (-31.74%) compared to 3f355ad 

I think we should increase that number again. I guess the main culprit is that `projects/atft/src/test.ts` doesn't run the demo in any way.

**Comment by ComFreek** [comment]
> Yesterday, found a workaround on internet (import not a library, but module via patch). During development it is nice to use auto-watch feature (when needed).

Ah, nice catch!

> Let's do in a separate Issue/PR.

Alright. You can already create an NPM user account if you don't have one yet.

I'll be off for today, have a nice evening!

**Comment by makimenko** [comment]
Thanks for a review and help!
Cheers!

**Pull request merged**